packages/gatsby-plugin-feed/src/internals.js

```diff
@@ -29,11 +29,13 @@ export const defaultOptions = {
   setup: ({
     query: {
       site: { siteMetadata },
-      ...rest
     },
+    generator,
+    ...rest
```

Comment:
So it looks like this does indeed fix the issue! Took a while for us to catch it so we need it now!